app/src/main/java/com/example/mystore/ui/profile/ProfileViewModel.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.mystore.ui.profile
+
+import androidx.lifecycle.ViewModel
+import com.example.mystore.TokenManager
+import com.example.mystore.repository.IDataStoreRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class ProfileViewModel @Inject constructor(
+    private val dataStoreRepository: IDataStoreRepository,
+    private val tokenManager: TokenManager,
+) : ViewModel() {
+
+    fun getEmail(): String {
+        return dataStoreRepository.getItem("email") ?: ""
+    }
+
+    fun logout(onLogout: () -> Unit = {}){
```

Comment:
Is this being used anywhere? Also, keep in mind that when you do use it, just navigating to `Login` won't suffice as you'll have to clear the backstack of all other opened screens up to this point as well. The documentation of the library is horrendous so you'll have to experiment (hint: you can test if you've popped the backstack or not when you introduce a temporary test TopBar to Login with a back button and try to `navigateUp` - the ideal result is you get thrown out of the app cause there are no screens in the backstack).

If you can't figure it out because of lack of documentation - ping us!